### PR TITLE
Add `UpdateParams` rpc and `Params` query to ibchooks module

### PR DESCRIPTION
**Pull request by nullpointer0x00**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

## Summary by CodeRabbit

- **New Features**
  - Introduced new query RPC endpoints (`UpdateParams` and `Params`) for `ibchooks`, `ibcratelimit`, `attribute`, and `marker` modules.
  - Added CLI commands for querying and updating parameters in the `ibchooks` module.

- **Bug Fixes**
  - Fixed integration tests for `ibchooks` to ensure accurate network parameter configurations and command testing.

- **Tests**
  - Added comprehensive test suites for `ibchooks` module, including tests for query parameters and update commands.

- **Documentation**
  - Updated `CHANGELOG.md` to reflect new RPC endpoints and CLI command enhancements.

### CODE REVIEW

File: x/ibchooks/client/cli/cli_test.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+		keyId := fmt.Sprintf("test_key%v", i)
+		info, _, err := kr.NewMnemonic(keyId, keyring.English, path, keyring.DefaultBIP39Passphrase, hd.Secp256k1)
+		s.Require().NoError(err, "Keyring.NewMneomonic")
+		addr, err := info.GetAddress()
+		if err != nil {
+			panic(err)
+		}
+		s.accountAddresses = append(s.accountAddresses, addr)
+	}
+}
```
Comment (SpicyLemon) [inline x/ibchooks/client/cli/cli_test.go:127]: Please use `testutil.GenerateTestKeyring` for this. See `x/attribute/client/cli/cli_test.go` for an example.

File: x/ibchooks/client/cli/cli_test.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+		genBalances = append(genBalances, banktypes.Balance{Address: s.accountAddresses[i].String(), Coins: sdk.NewCoins(
+			sdk.NewInt64Coin("nhash", 100_000_000), sdk.NewInt64Coin(s.cfg.BondDenom, 100_000_000),
+		).Sort()})
+	}
+	var bankGenState banktypes.GenesisState
+	bankGenState.Params = banktypes.DefaultParams()
+	bankGenState.Balances = genBalances
+	bankDataBz, err := s.cfg.Codec.MarshalJSON(&bankGenState)
+	s.Require().NoError(err, "should be able to marshal bank genesis state when setting up suite")
+	genesisState[banktypes.ModuleName] = bankDataBz
```
Comment (SpicyLemon) [inline x/ibchooks/client/cli/cli_test.go:74]: Please use `testutil.MutateGenesisState` for this stuff.

File: x/ibchooks/types/msgs.go [hunk 3]
```diff
@@ -1,15 +1,43 @@
...
+		},
+		Authority: authority,
+	}
+}
+
+// ValidateBasic implements the sdk.Msg interface.
+func (msg MsgUpdateParamsRequest) ValidateBasic() error {
+	for _, contract := range msg.Params.AllowedAsyncAckContracts {
+		if _, err := sdk.AccAddressFromBech32(contract); err != nil {
+			return fmt.Errorf("invalid contract address: %s", contract)
```
Comment (SpicyLemon) [inline x/ibchooks/types/msgs.go:34]:
```suggestion
			return fmt.Errorf("invalid contract address %q: %w", contract, err)
```

**Review by Taztingo — APPROVED** [review]

File: x/ibchooks/client/cli/cli_test.go (hunk 2 above)
Comment (nullpointer0x00) [reply x/ibchooks/client/cli/cli_test.go]: Oh yes, so much cleaner!  Copy and pasted this setup from somewhere else...  We should probably make an issue (`nice-to-have`) to clean this up to all use the `testutil.MutateGenesisState`.

**Review by SpicyLemon — APPROVED** [review]

**Pull request merged**